tests/unit/test_chaperone.js

```diff
@@ -0,0 +1,57 @@
+const path				= require('path');
+const log				= require('@whi/stdlog')(path.basename( __filename ), {
+    level: process.env.LOG_LEVEL || 'fatal',
+});
+
+const expect				= require('chai').expect;
+
+const {}				= require("../mock_browser_context.js");
+const { setup_mock_fetch }		= require("../mock_browser_context.js");
+const mock_envoy_server			= require("../mock_envoy.js");
+
+const { Chaperone }			= require("../../build/index.js");
+
+const wait = require('waait')
+
+let server, chaperone;
+
+describe('chaperone.ready', () => {
+  before("Setup fetch mocks", () => {
+    setup_mock_fetch();
+  })
+
+  after("Shutdown mock server", async function () {
+    log.info("Waiting for Chaperone to close");
+    await chaperone.close();
+  
+    log.info("Waiting for server to close");
+    await server.close();
+  });
+
+  it('waits for connection with server before returning', async () => {
+    chaperone			= new Chaperone({
+	    "comb": false,
+	    "timeout": 2000,
+	    "debug": !!process.env.LOG_LEVEL,
+	    "connection": {
+    		"port": 4656,
+      }
+    });
+
+    const startServerWithDelay = async () => {
+      await wait(500)
+      log.info('Starting server')
+      server				= mock_envoy_server();
+    }
+
+    const startChaperoneAndServer = () => {
+      startServerWithDelay();
+      return chaperone.ready( 2000 );
+    }
+
+    log.info("Waiting for Chaperone to be ready...");
+    await startChaperoneAndServer();
+    log.info("Chaperone is ready");
+    expect( chaperone.opened ).to.be.true;
```

Comment:
Here is how it could be done without the server just as a unit test
```suggestion
      await wait(); // queue after async init                                                                                                                                                                                                                                                                                                                                                                                            

      chaperone.conn            = {};
      chaperone.opened          = true;
      await chaperone.ready( 1000 );

      chaperone.opened          = false;
      let failed                = false;
      try {
          log.info("Waiting for Chaperone to be ready...");
          await chaperone.ready( 1000 );
          log.info("Chaperone is ready");
      } catch ( err ) {
          failed                = true;
          expect( err.name ).to.equal("TimeoutError");
      }

      expect( failed ).to.be.true;
```